beacon-chain/sync/subscriber.go

```diff
@@ -517,20 +533,13 @@ func (s *Service) pruneSubscriptions(t *subnetTracker, wantedSubnets map[uint64]
 // subscribeToSubnets subscribes to needed subnets and unsubscribe from unneeded ones.
 // This functions mutates the `subscriptionBySubnet` map, which is used to keep track of the current subscriptions.
 func (s *Service) subscribeToSubnets(t *subnetTracker) error {
-	// Do not subscribe if not synced.
-	if s.chainStarted.IsSet() && s.cfg.initialSync.Syncing() {
+	select {
+	default:
+		// If initialSyncComplete is not closed, bail out; subscribeWithParameters will keep calling this
```

Comment:
Couln't we also wait for initial sync to be completed and immediately subscribe to subnets?
It would prevent a potential 12 seconds after initial sync completion without any subscription, and so some avoidable additional Req/Resp calls.